OptimizelySDK.Tests/OptimizelyTest.cs

```diff
@@ -1109,5 +1120,350 @@ public void TestGetVariationBucketingIdAttribute()
+                It.IsAny<UserAttributes>(), featureVariableType)).Returns("Test String");
+            OptimizelyMock.Setup(om => om.GetFeatureVariableValueForType(It.IsAny<string>(), variableKeyIntString, It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), featureVariableType)).Returns("123");
+            OptimizelyMock.Setup(om => om.GetFeatureVariableValueForType(It.IsAny<string>(), variableKeyNull, It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), featureVariableType)).Returns<string>(null);
+
+            Assert.AreEqual("Test String", OptimizelyMock.Object.GetFeatureVariableString(featureKey, variableKeyString, TestUserId, null));
+            Assert.AreEqual("123", OptimizelyMock.Object.GetFeatureVariableString(featureKey, variableKeyIntString, TestUserId, null));
+            Assert.AreEqual(null, OptimizelyMock.Object.GetFeatureVariableString(featureKey, variableKeyNull, TestUserId, null));
+        }
+
+        #endregion // Test GetFeatureVariable<Type> TypeCasting
+
+        #region Test GetFeatureVariableValueForType method
+
+        // Should return null and log error message when arguments are null or empty.
+        [Test]
+        public void TestGetFeatureVariableValueForTypeGivenNullOrEmptyArguments()
+        {
+            var featureKey = "featureKey";
+            var variableKey = "variableKey";
+            var variableType = FeatureVariable.VariableType.BOOLEAN;
+
+            Assert.IsNull(Optimizely.GetFeatureVariableValueForType(null, variableKey, TestUserId, null, variableType));
+            Assert.IsNull(Optimizely.GetFeatureVariableValueForType("", variableKey, TestUserId, null, variableType));
+
+            Assert.IsNull(Optimizely.GetFeatureVariableValueForType(featureKey, null, TestUserId, null, variableType));
+            Assert.IsNull(Optimizely.GetFeatureVariableValueForType(featureKey, "", TestUserId, null, variableType));
+
+            Assert.IsNull(Optimizely.GetFeatureVariableValueForType(featureKey, variableKey, null, null, variableType));
+            Assert.IsNull(Optimizely.GetFeatureVariableValueForType(featureKey, variableKey, "", null, variableType));
+
+            LoggerMock.Verify(l => l.Log(LogLevel.ERROR, "Feature flag key must not be empty."), Times.Exactly(2));
+            LoggerMock.Verify(l => l.Log(LogLevel.ERROR, "Variable key must not be empty."), Times.Exactly(2));
+            LoggerMock.Verify(l => l.Log(LogLevel.ERROR, "User ID must not be empty."), Times.Exactly(2));
+        }
+
+        // Should return null and log error message when feature key or variable key does not get found.
+        [Test]
+        public void TestGetFeatureVariableValueForTypeGivenFeatureKeyOrVariableKeyNotFound()
+        {
+            var featureKey = "this_feature_should_never_be_found_in_the_datafile_unless_the_datafile_creator_got_insane";
```

Comment:
feature_key_not_found

Reply:
Anything bad in this? this_feature_should_never_be_found_in_the_datafile_unless_the_datafile_creator_got_insane
:)